### PR TITLE
✨  Move lowestNonZeroRetryAfterSeconds to util

### LINKED ISSUE #6945: Consider dedup lowestNonZeroRetryAfterSeconds
During the Runtime SDK implementation we introduced a copy of `lowestNonZeroRetryAfterSeconds`.

Let's consider if we should deduplicate and only have this once as a util.

/kind cleanup
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by LuBingtan**
**What this PR does / why we need it**:
Fix https://github.com/kubernetes-sigs/cluster-api/issues/6945

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6945

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

File: util/util.go [hunk 1]
```diff
@@ -593,6 +593,20 @@ func LowestNonZeroResult(i, j ctrl.Result) ctrl.Result {
 	}
 }
 
+// LowestNonZeroRetryAfterSeconds returns the lowest non-zero value of the two provided values.
+func LowestNonZeroRetryAfterSeconds(i, j int32) int32 {
```
Comment (vincepri) [inline util/util.go:597]:
```suggestion
// LowestNonZero returns the lowest non-zero value of the two provided values.
func LowestNonZero(i, j int32) int32 {
```

Given that's just int32?

Comment (LuBingtan) [reply util/util.go]: Thanks for review 😄 
Yeah, right now the function is just used to compare the return value of `GetRetryAfterSeconds`, and the type of return value is int32.

And the suggestion on name change makes sense. Already updated

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**